holidays/groups/thai.py

```diff
@@ -0,0 +1,172 @@
+#  python-holidays
+#  ---------------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: dr-prodigy <maurizio.montel@gmail.com> (c) 2017-2022
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/dr-prodigy/python-holidays
+#  License: MIT (see LICENSE file)
+
+from datetime import date
+from typing import Optional
+
+from holidays.calendars import _ThaiLunisolar, KHMER_CALENDAR, THAI_CALENDAR
+
+
+class ThaiCalendarHolidays:
+    """
+    Thai lunisolar calendar holidays.
+
+    For more info, see class `_ThaiLunisolar`.
+    """
+
+    def __init__(self, calendar=THAI_CALENDAR) -> None:
+        self.__verify_calendar(calendar)
+        self.__calendar = calendar
+        self.__thai_calendar = _ThaiLunisolar(calendar)
+
+    @staticmethod
+    def __verify_calendar(calendar):
+        """
+        Verify calendar type.
+        """
+        if calendar not in {KHMER_CALENDAR, THAI_CALENDAR}:
+            raise ValueError(
+                f"Unknown calendar name: {calendar}. "
+                "Use `KHMER_CALENDAR` or `THAI_CALENDAR`."
+            )
+
+    def _add_asarnha_bucha(self, name, calendar=None) -> Optional[date]:
+        """
+        Add Asarnha Bucha.
+
+        Asalha Pūjā (also written as Asarnha Bucha Day) is a Buddhist festival
+        celebrated on the 15th Waxing Day (Full Moon) of Month 8.
+
+        Khmer variant: always fall on Month 8.
+        Thai variant:  will use Month 8.8 instead for Athikamat years.
+
+        https://en.wikipedia.org/wiki/Asalha_Puja
+        """
+        calendar = calendar or self.__calendar
+        self.__verify_calendar(calendar)
```

Comment:
I think, there is no need to call `__verify_calendar()`, since it's called in `asarnha_bucha_date()` anyway. 👀
Same for similar functions below.

Reply:
Redundant checks are removed now, fixed.